### PR TITLE
GitHub Actions linter for Rust formatting

### PR DESCRIPTION
**Pull request by duthils**
This PR takes the changes from https://github.com/getsops/sops/pull/1403 and applies them in a dedicated CI workflow.

Fixes https://github.com/getsops/sops/pull/1390

### CODE REVIEW

File: .github/workflows/linters.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  contents: read
+
+jobs:
+  lint:
+    name: Lint Rust source files
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
```
Comment (felixfontein) [inline .github/workflows/linters.yml:24]: I think we should also install rust here. I've copied this from `.github/workflows/cli.yml`:
```suggestion

      - name: Install rustup
        run: curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | bash -s -- -y --default-toolchain 1.70.0

```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution!

**Comment by duthils** [comment]
Rust is already available on [Github runners in version 1.80.1](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md#rust-tools), but I guess we want to keep the same version as is used for running the tests.

Thanks for the suggestion!

**Comment by felixfontein** [comment]
> Rust is already available on [Github runners in version 1.80.1](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md#rust-tools), but I guess we want to keep the same version as is used for running the tests.

Yep; also having a controlled environment is generally a good idea :) Then we can decide which Rust version to use, instead of having to use the version provided by the GHA image.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@duthils @shikharvashistha thanks for your contribution!

**Pull request merged**